### PR TITLE
[export] Allow constant attr mutation

### PR DESCRIPTION
**Pull request by angelayi**
Test Plan: CI

Differential Revision: D56893728

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @chauhang

### CODE REVIEW

**Review by pianpwk — APPROVED** [review]

File: torch/_dynamo/symbolic_convert.py [hunk 1]
```diff
@@ -2498,7 +2498,11 @@ def inline_call_(
             args
             and isinstance(
                 args[0],
-                (variables.CustomizedDictVariable, variables.UserDefinedObjectVariable),
+                (
+                    variables.CustomizedDictVariable,
+                    variables.UserDefinedObjectVariable,
+                    variables.NNModuleVariable,
```
Comment (avikchaudhuri) [inline torch/_dynamo/symbolic_convert.py:2504]: What does this do? It's hard to wrap my head around the condition, but is it that you can now define `__setattr__` on a `nn.module`? 

If so does it get inlined? 

Why was it banned before?

File: test/dynamo/test_export_mutations.py [hunk 2]
```diff
@@ -57,18 +44,6 @@ def forward(self, x):
 
         self.check_failure_on_export(Foo(), torch.randn(3, 2))
 
-    def test_module_attribute_mutation_violation_positive_4(self):
```
Comment (avikchaudhuri) [inline test/dynamo/test_export_mutations.py:60]: So this works now? Just gets inlined into a constant on the RHS, is that why? 

But I don't see `self.a` being initialized, so isn't that a new attribute, like `test_module_attribute_mutation_violation_positive_3`? Confused about the mental model here.

File: torch/_dynamo/symbolic_convert.py (hunk 1 above)
Comment (angelayi) [reply torch/_dynamo/symbolic_convert.py]: Ah sorry, I think this was part of another direction I was trying to take to fix the const attr issue -- I deleted this change and looks like the tests still work 😅

File: test/dynamo/test_export_mutations.py (hunk 2 above)
Comment (angelayi) [reply test/dynamo/test_export_mutations.py:60]: Yeah, self.a is a new attribute, but it's a constant like in `mutation_violation_positive_3`. `mutation_violation_positive_3` is failing due to a different error, so I moved it to `test_export` instead.

**Pull request closed without merging**